src/client/network/mqtt/mqtt_mosquitto.c

```diff
@@ -0,0 +1,196 @@
+// Copyright 2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+#include <mosquitto.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include "client/network/mqtt/mqtt.h"
+
+struct mqtt_client {
+  struct mosquitto *mosq;
+  mqtt_client_config_t *config;
+  void (*mqtt_callback)(mqtt_client_event_t *event, void *userdata);
+  void *cb_userdata;
+};
```

Comment:
My idea is that, struct mqtt_client will depend on the mqtt client. 
For eg now since we use mosquitto as mqtt client for linux we have struct mosquitto *mosq declared in struct mqtt_client in file mqtt_mosquitto.c.
For esp32 similarly esp_mqtt_client_handle_t client will be declared in struct mqtt_client in file mqtt_esp32.c.